src/main.rs

```diff
@@ -95,29 +94,35 @@ fn parse_args() -> anyhow::Result<MacheteArgs> {
     })
 }
 
-fn collect_paths(path: &Path, skip_target_dir: bool) -> Vec<PathBuf> {
-    // Find directory entries.
-    let walker = WalkDir::new(path).into_iter();
+fn is_target_dir(entry: &ignore::DirEntry) -> bool {
+    entry.file_type().map(|t| t.is_dir()).unwrap_or_default() && entry.file_name() == "target"
+}
 
-    let manifest_path_entries = if skip_target_dir {
-        walker
-            .filter_entry(|entry| !entry.path().ends_with("target"))
-            .collect()
-    } else {
-        walker.collect::<Vec<_>>()
-    };
+fn is_cargo_toml_file(entry: &ignore::DirEntry) -> bool {
+    entry.file_type().map(|t| t.is_file()).unwrap_or_default() && entry.file_name() == "Cargo.toml"
+}
 
-    manifest_path_entries
-        .into_iter()
-        .filter_map(|entry| match entry {
-            Ok(entry) if entry.file_name() == "Cargo.toml" => Some(entry.into_path()),
-            Err(err) => {
-                eprintln!("error when walking over subdirectories: {}", err);
-                None
-            }
-            _ => None,
-        })
-        .collect()
+fn build_toml_file_iterator(
+    path: &Path,
+    skip_target_dir: bool,
+) -> impl Iterator<Item = Result<ignore::DirEntry, ignore::Error>> {
+    let mut walk_builder = ignore::WalkBuilder::new(path);
+
+    // NOTE(mickvangelderen): This makes the skip_target_dir = false test pass, but I am not sure
+    // this is how we want to implement this. Perhaps we should add an argument to toggle usage of
+    // ignore files and modify the tests.
+    walk_builder.git_exclude(false);
+    walk_builder.git_global(false);
+    walk_builder.git_ignore(false);
```

Comment:
This makes the skip_target_dir = false test pass, but I am not sure
this is how we want to implement this. Perhaps we should add an argument to toggle usage of
ignore files and modify the tests.

Reply:
I'd like to have a look at `cargo udeps` to see what excluding/filtering/ignoring options they provide, we may be able to draw some inspiration from a more mature project.

Reply:
Hey hey! sorry for the need to rebase, but now with argument parsing with `argh` it might be simpler to add new options, at least.

Reply:
> I'd like to have a look at cargo udeps to see what excluding/filtering/ignoring options they provide

They just have a `--exclude <pkgid>` flag, which is passed to `cargo check`. 

Take `workspace-hack` as an example, it won't check `workspace-hack`'s dependencies, but other crate's dependency to `workspace-hack` is still viewed as unused. 🤔